aptos-move/move-examples/mint_nft_v2/4-Getting-Production-Ready/sources/create_nft_getting_production_ready.move

```diff
@@ -0,0 +1,478 @@
+            token_uri,
+            property_keys: vector<String>[string::utf8(b"given_to")],
+            property_types: vector<String>[ string::utf8(b"address") ],
+            property_values: vector<vector<u8>>[bcs::to_bytes(&@source_addr)],
+            minting_enabled: true,
+            expiration_timestamp: 10000000000,
+            token_minting_events: account::new_event_handle<TokenMintingEvent>(resource_signer),
+        });
+
+        //Store the collection object address with a currently empty most recently minted token object address for later.
+        move_to(resource_signer, ObjectAddresses {
+            collection_object_address,
+            last_minted_token_object_address: option::none(),
+        });
+    }
+
+    /// Mint an NFT to the receiver.
+    /// `mint_proof_signature` should be the `MintProofChallenge` signed by the admin's private key
+    /// `public_key_bytes` should be the public key of the admin
+    public entry fun mint_event_ticket(receiver: &signer, mint_proof_signature: vector<u8>) acquires ModuleData, ObjectAddresses {
+        // get the collection minter and check if the collection minting is disabled or expired
+        let module_data = borrow_global_mut<ModuleData>(@mint_nft_v2);
+        assert!(timestamp::now_seconds() < module_data.expiration_timestamp, error::permission_denied(ECOLLECTION_EXPIRED));
+        assert!(module_data.minting_enabled, error::permission_denied(EMINTING_DISABLED));
+
+        let receiver_addr = signer::address_of(receiver);
+        // verify that the `mint_proof_signature` is valid against the admin's public key
+        verify_proof_of_knowledge(receiver_addr,
+            mint_proof_signature,
+            module_data.collection_name,
+            @mint_nft_v2,
+            module_data.token_name,
+            module_data.public_key);
+
+        // mint token to the receiver
+        let resource_signer = account::create_signer_with_capability(&module_data.signer_cap);
+        let resource_address = signer::address_of(&resource_signer);
+        let token_creation_num = account::get_guid_next_creation_num(resource_address);
+
+        aptos_token::mint(
+            &resource_signer,
+            module_data.collection_name,
+            module_data.token_description,
+            module_data.token_name,
+            module_data.token_uri,
+            module_data.property_keys,
+            module_data.property_types,
+            module_data.property_values,
+        );
+
+        let token_object_address = object::create_guid_object_address(resource_address, token_creation_num);
+        let token_object = object::address_to_object<AptosToken>(object::create_guid_object_address(resource_address, token_creation_num));
+        object::transfer(&resource_signer, token_object, receiver_addr);
+
+        let object_addresses = borrow_global_mut<ObjectAddresses>(@mint_nft_v2);
+        if (option::is_some(&object_addresses.last_minted_token_object_address)) {
```

Comment:
Do you know what last_minted_token_object_address  is used for?

Reply:
I was using it to view it myself, meant to take it out, hah